lib/credentials.js

```diff
@@ -131,6 +125,40 @@ Credentials.prototype.hasWalletInfo = function() {
   return !!this.walletId;
 };
 
+Credentials.prototype.isPrivKeyEncrypted = function() {
+  return (!!this.xPrivKeyEnc) && !this.xPrivKey;
+};
+
+Credentials.prototype.hasPrivKeyEncrypted = function() {
+  return (!!this.xPrivKeyEnc);
+};
+
+Credentials.prototype.encrypt = function(password, opts) {
+  if (this.xPrivKeyEnc)
+    throw new Error('Encrypted Privkey Already exists');
+
+  if (!this.xPrivKey)
+    throw new Error('No private key to encrypt');
+
+
+  this.xPrivKeyEnc = sjcl.encrypt(password, this.xPrivKey, opts);
+  if (!this.xPrivKeyEnc)
+    throw new Error('Could not encrypt');
+};
+
+Credentials.prototype.lock = function() {
```

Comment:
There are 3 methods:
 - encrypt: => Encrypt for the first time. 
 - unlock => decrypt and keep it decrypted 
 - lock =>  remove the decrypted key

The idea is that you can `unlock(password)` and to several operations before 'locking' again; and that locking does not require a password.